### PR TITLE
[fix](nereids)left outer join estimation

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
A left join B on A.x=B.x and A.y=B.y
B.x and B.y make result tuple number scale out.
suppose A is scaled out by B.x N1 time, and scaled out by B.y N2 time, and N1 < N2.
we should choose N1 as the final scale out factor, not N2.

this pr impact on tpcds_sf100 59/17/29/25/47/40/54 

before
query59 77295   75279   75230   75230
query17 22642   21566   21599   21566
query29 16508   16092   16006   16006
query25 20262   20571   21171   20571
query47 23571   23264   23107   23107
query40 3305    2849    3064    2849
query54 9052    8882    8715    8715
Total cold run time: 172635 ms
Total hot run time: 168044 ms

after 
query59 56435   54717   53919   53919
query17 24167   22377   23237   22377
query29 16950   18325   16333   16333
query25 21478   22975   21358   21358
query47 24412   24611   23920   23920
query40 5491    4779    5176    4779
query54 8671    8664    8658    8658
Total cold run time: 157604 ms
Total hot run time: 151344 ms

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/JoinEstimation.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ private static Statistics estimateHashJoin(Statistics leftStats, Statistics righ
             outputRowCount = outputRowCount * Math.pow(0.9, unTrustableCondition.size());
         } else {
             outputRowCount = Math.max(leftStats.getRowCount(), rightStats.getRowCount());
-            Optional<Double> ratio = unTrustEqualRatio.stream().max(Double::compareTo);
+            Optional<Double> ratio = unTrustEqualRatio.stream().min(Double::compareTo);
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/JoinEstimation.java:140]: would better to make a policy for mutiple column ndv estimation based on reversed order, similar with policy in gby?

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/JoinEstimation.java:140]: there are some difference. consider two condition: A left join B on A.x=B.x and A.y=B.y
suppose
A left join B on A.x=B.x, output expand A by 3 times,
A left join B on A.y=B.y, output expand A by 5 times.

‘ A left join B on A.x=B.x and A.y=B.y’ output should be less than ‘A left join B on A.x=B.x’

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Pull request merged**